intTests/test_mir_verify_structs/test.saw

```diff
@@ -0,0 +1,98 @@
+enable_experimental;
+
+m <- mir_load_module "test.linked-mir.json";
+
+let s1_adt = mir_find_adt m "test::S1" [];
+let s2_adt = mir_find_adt m "test::S2" [mir_u32, mir_u32];
+let s3_adt = mir_find_adt m "test::S3" [];
+let s4_adt = mir_find_adt m "test::S4" [];
```

Comment:
I'm mainly wondering whether `mir_find_adt` will fail if the type parameters get too complex.  It looks like the implementation is based on `Mir.Generator.findAdtInst`, which is only used in a few `Mir.TransCustom` cases and probably hasn't been thoroughly tested.  In particular, I'm not sure about the use of `==` on `TyAdt`, since I know crux-mir internally generates (or at least used to generate) fake `TyAdt`s where certain fields have bogus values.

Any generic type `Foo<T>` would work in place of `Option` for this.  Though I would expect `mir_find_adt` to successfully look up `Option<_>`, even though you can't use the resulting ADT for anything at the moment.

Reply:
Ah, thanks for clarifying. I'll add some test cases for things like `Foo<[u8; 4]>` and `Foo<test::S2<u32, u32>>`, where `Foo` is a struct. I'll also add an example of using `mir_find_adt` to look up `Option<_>` (but without using it in an actual specification).